src/main/scala/uk/gov/nationalarchives/notifications/decoders/DiskSpaceAlarmDecoder.scala

```diff
@@ -0,0 +1,19 @@
+package uk.gov.nationalarchives.notifications.decoders
+
+import io.circe.{Decoder, HCursor}
+import io.circe.generic.auto._
+import uk.gov.nationalarchives.notifications.decoders.IncomingEvent.{Record, parseSNSMessage}
+
+object DiskSpaceAlarmDecoder {
+
+  case class AlarmDimension(name: String, `value`: String)
+  case class AlarmTrigger(Dimensions: List[AlarmDimension], Threshold: Int)
+  case class DiskSpaceAlarmEvent(NewStateValue: String, AlarmName: String, Trigger: AlarmTrigger) extends IncomingEvent
+
+  val decodeDiskSpaceAlertEvent: Decoder[IncomingEvent] = (c: HCursor) => for {
+    messages <- c.downField("Records").as[List[Record]]
```

Comment:
Potentially could abstract out the function to retrieve the "messages", as I think its the same for Keycloak events  as well.

Reply:
I've abstracted it out into the `IncomingEvent` object and just called it inside that object instead of creating `decodeExportStatusEvent`, `decodeKeycloakEvent` and `decodeDiskSpaceAlertEvent` in the other classes. That is better. 

`DiskSpaceAlarmDecoder`, `ExportStatusDecoder` and `KeycloakEventDecoder` now only contain the case classes for the decoding and I'm wondering whether it's worth having them there for that or if it might be better to move the case classes into `IncomingEvent` and delete those three classes. What do you think?